controller/linodemachine_controller.go

```diff
@@ -308,6 +310,10 @@ func (r *LinodeMachineReconciler) reconcileCreate(
 
 		linodeInstance, err = machineScope.LinodeClient.CreateInstance(ctx, *createOpts)
 		if err != nil {
+			if linodego.ErrHasStatus(err, linodeTooManyRequests) || linodego.ErrHasStatus(err, linodeEOFOrTimeout) {
```

Comment:
Also check if `linodego.ErrorFromError` works here or not instead of using `linodeEOFOrTimeout`